tests/pubsub/utils.nim

```diff
@@ -8,10 +8,10 @@ proc generateNodes*(num: Natural, gossip: bool = false): seq[Switch] =
     result.add(newStandardSwitch(gossip = gossip))
 
 proc subscribeNodes*(nodes: seq[Switch]) {.async.} =
-  var dials: seq[Future[void]]
+  var dials: seq[Future[Connection]]
   for dialer in nodes:
     for node in nodes:
       if dialer.peerInfo.peerId != node.peerInfo.peerId:
-        dials.add(dialer.subscribeToPeer(node.peerInfo))
+        dials.add(dialer.dial(node.peerInfo))
   await sleepAsync(100.millis)
```

Comment:
I see useless `sleepAsync` here.

Reply:
this gives time fo the two peers to establish the tcp connection.

Reply:
All required time will be consumed in next call `await allFutures` you don't need to put sleepAsync here.